src/services/TestResultsService.ts

```diff
@@ -283,10 +283,14 @@ export class TestResultsService {
                       payload.testTypes[index] = testType;
                     }
                   } else if (payload.vehicleType === VEHICLE_TYPES.HGV || payload.vehicleType === VEHICLE_TYPES.TRL) {
-                    // Checks for testType = First test or First test Retest AND if vehicle doesn't currently have an existing expiry date, i.e set to default
-                    if (this.isFirstTestRetestTestType(testType) && dateFns.isEqual(mostRecentExpiryDateOnAllTestTypesByVin, new Date(1970, 1, 1))) {
-                      // Applying CVSB-8658 logic changes if vehicle doesn't currently have an existing expiry date
-                      const regOrFirstUseDate = payload.vehicleType === VEHICLE_TYPES.HGV ? payload.regnDate : payload.firstUseDate;
+                    // Applying CVSB-8658 logic changes if vehicle doesn't currently have an existing expiry date
+                    const regOrFirstUseDate = payload.vehicleType === VEHICLE_TYPES.HGV ? payload.regnDate : payload.firstUseDate;
+                    // preparaing compare date for CVSB-9187 to compare first test/retest conducted after anniversary date
+                    const firstTestAfterAnvCompareDate = dateFns.addYears(dateFns.startOfMonth(regOrFirstUseDate), 1);
+                    // Checks for testType = First test or First test Retest AND if firstUse or registraion date is 1 year from the month of first use.
+                    if (this.isFirstTestRetestTestType(testType) && dateFns.isAfter(new Date(), firstTestAfterAnvCompareDate)) {
```

Comment:
The comment at the line above says that the condition should check if firstUse or registration date is 1 year from the month of first use. But here I see it compared with new Date(). Is that correct?

Reply:
typo in comment. It should be test date is 1 year from the month of first use or registration date.